### PR TITLE
Create sidebar components for editing and managing nodes

### PR DESCRIPTION
**Pull request by jtomeck**
Create necessary components for node and edge management. This group of sidebar tools allows you to search for node type and value, as well as edge type, operator and values.

### CODE REVIEW

File: src/components/Controls.vue [hunk 1]
```diff
@@ -17,6 +17,8 @@ export default Vue.extend({
     networkName: string,
     selectNeighbors: boolean,
     visualizedAttributes: string[],
+    nodeEditor: boolean,
+    connectivity: any,
```
Comment (waxlamp) [inline src/components/Controls.vue:21]: Can we use a better type than `any`?

File: src/components/Controls.vue [hunk 2]
```diff
@@ -66,63 +84,376 @@ export default Vue.extend({
...
+                            ></v-select>
+                          </v-col>
+                        </v-row>
+                      </v-expansion-panel-content>
+                    </v-expansion-panel>
+                  </v-expansion-panels>
+                  <v-divider />
+                </v-card-text>
+                <v-card-actions class="pa-3">
+                  <!-- DERYA -->
```
Comment (waxlamp) [inline src/components/Controls.vue:314]:
```suggestion
```

**Review by waxlamp — CHANGES_REQUESTED** [review]
Thanks for everyone's efforts on this branch.

I just had one suggestion to tighten up a type, and I think one more direct code suggestion (to get rid of a comment that we don't need).

File: src/components/Controls.vue (hunk 2 above)
Comment (waxlamp) [reply src/components/Controls.vue:314]: @gotdairyya, could you accept this code suggestion?

File: src/components/Controls.vue (hunk 1 above)
Comment (waxlamp) [reply src/components/Controls.vue:21]: @gotdairyya, let me know if you'd like me to address this? I can probably figure out what type to put in here.

Comment (gotdairyya) [reply src/components/Controls.vue:21]: @waxlamp Is any ok for now? Connectivity is a fill-in variable to hold the dropdown menu items, but if we refactor the sidebar as a form then I'm not sure we will need it

Comment (waxlamp) [reply src/components/Controls.vue:21]: Ah I see. I am surprised the linter doesn't complain about it. Yes this is ok for now. Could you add a comment above that line to explain what you just told me?

**Review by waxlamp — APPROVED** [review]

**Comment by waxlamp** [comment]
I am working on resolving the conflicts in this PR--watch for a merge commit that does so.

**Comment by waxlamp** [comment]
> I am working on resolving the conflicts in this PR--watch for a merge commit that does so.

This is done now, but I am not sure I resolved the conflicts properly. @jtomeck, if you could pull and build and verify this looks ok, we can merge.

**Review by JackWilb — APPROVED** [review]
Since this is going into the graffinity branch, it's okay that we're losing the settings toggles. We should be aware that we'll need a spot for them at some point though

**Comment by waxlamp** [comment]
> Since this is going into the graffinity branch, it's okay that we're losing the settings toggles. We should be aware that we'll need a spot for them at some point though

Would you mind checking out an earlier revision from this PR branch and see if the controls are gone there too? If not, then I likely made a mistake in the merge commit.

**Comment by JackWilb** [comment]
> > Since this is going into the graffinity branch, it's okay that we're losing the settings toggles. We should be aware that we'll need a spot for them at some point though
> 
> Would you mind checking out an earlier revision from this PR branch and see if the controls are gone there too? If not, then I likely made a mistake in the merge commit.

I remember them being removed in the earlier commits from Jared. We had talked about relocating them

**Comment by waxlamp** [comment]
> > > Since this is going into the graffinity branch, it's okay that we're losing the settings toggles. We should be aware that we'll need a spot for them at some point though
> > 
> > 
> > Would you mind checking out an earlier revision from this PR branch and see if the controls are gone there too? If not, then I likely made a mistake in the merge commit.
> 
> I remember them being removed in the earlier commits from Jared. We had talked about relocating them

Thanks, Jack. @jtomeck, please give this a look as soon as you can, and if you're happy with it, please go ahead and merge it.

**Pull request merged**